packages/interactivity/src/utils.ts

```diff
@@ -20,9 +20,16 @@ import {
 	setNamespace,
 	resetNamespace,
 } from './hooks';
+import type { Flusher, ParentElement } from './types';
 
-const afterNextFrame = ( callback ) => {
-	return new Promise( ( resolve ) => {
+/**
+ * Executes a callback function after the next frame is rendered.
+ *
+ * @param callback - The callback function to be executed.
+ * @return A promise that resolves after the callback function is executed.
+ */
+const afterNextFrame = ( callback: Function ) => {
```

Comment:
`Function` isn't an ideal type for this, [it's better to express the callback type](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#callback-types):

```suggestion
const afterNextFrame = ( callback: () => void ) => {
```